Microsoft.Toolkit.Uwp.UI.Controls/AdaptiveGridView/AdaptiveGridView.cs

```diff
@@ -100,6 +100,13 @@ protected override void PrepareContainerForItemOverride(DependencyObject obj, ob
         /// <returns>The calculated item width.</returns>
         protected virtual double CalculateItemWidth(double containerWidth)
         {
+            // quick check for containerWidth to avoid cyclic cases where size
```

Comment:
I think it would be best to return double.Nan if the DesiredWidth has not been set. This allows the items to take up the size required and allow the GridView to do the layout of items. This works fine when used with a horiztonal scrolling scrollViewer but does have rendering/graphics issues with the Bladeview. However, the issues I see with the AdaptiveGridView I also see with the GridView so I think it might be a core issue with rendering.

Reply:
To clarify, I think the first if should be changed to:
```
            if (double.IsNaN(DesiredWidth))
            {
                return DesiredWidth;
            }

            var columns = CalculateColumns(containerWidth, DesiredWidth);
```

Reply:
Much better solution, works great. I just changed it :)